src/python/grpcio/grpc/_cython/_cygrpc/records.pyx.pxi

```diff
@@ -303,21 +304,46 @@ cdef class SslPemKeyCertPair:
     self.c_pair.certificate_chain = self.certificate_chain
 
 
+
+cdef void* copy_ptr(void* ptr):
+  return ptr
+
+
+cdef void destroy_ptr(void* ptr):
+  pass
+
+
+cdef int compare_ptr(void* a, void* b):
+  if a < b:
+    return -1
+  elif a > b:
+    return 1
+  else:
+    return 0
+
+
 cdef class ChannelArg:
 
   def __cinit__(self, bytes key, value):
     self.key = key
+    self.value = value
     self.c_arg.key = self.key
     if isinstance(value, int):
-      self.value = value
       self.c_arg.type = GRPC_ARG_INTEGER
       self.c_arg.value.integer = self.value
     elif isinstance(value, bytes):
-      self.value = value
       self.c_arg.type = GRPC_ARG_STRING
       self.c_arg.value.string = self.value
     else:
-      raise TypeError('Expected int or bytes, got {}'.format(type(value)))
+      # Pointer objects must override __int__() to return
+      # the underlying C address.  This is how SWIG exposes
+      # pointers (Python ints are word size)
```

Comment:
I would strongly suggest further that the pointee's lifetime be explicitly documented as bound to the lifetime of the Python object. That way the vtable copy/destroy methods can be meaningfully used w/ e.g. `Py_INCREF` to ensure we don't cause dangling references.